FileConv/Gfx/FontrixFont.cs

```diff
@@ -0,0 +1,506 @@
+                if (FileAttrs.AuxType == 0x6400) {
+
+                    // Only need to read up to ID_BYTE_3 to validate. +1 to get size from offset.
+                    var buffer = new byte[ID_BYTE_3 + 1];
+
+                    DataStream.Position = 0;
+                    DataStream.ReadExactly(buffer, 0, ID_BYTE_3 + 1);
+
+                    bool valid = buffer[ID_BYTE_1] == ID_BYTE_1_VAL &&
+                                 buffer[ID_BYTE_2] == ID_BYTE_2_VAL &&
+                                 buffer[ID_BYTE_3] == ID_BYTE_3_VAL &&
+                                 buffer[CHAR_COUNT_LOCATION] <= MAX_NUM_CHARACTERS;
+
+                    if (valid) {
+                        return Applicability.Yes;
+                    } else {
+                        Debug.WriteLine("Valid is not set for this font.");
+                    }
+                }
+            }
+            return Applicability.Not;
+        }
+
+        public override Type GetExpectedType(Dictionary<string, string> options) {
+            return typeof(IBitmap);
+        }
+
+        public override IConvOutput ConvertFile(Dictionary<string, string> options) {
+            if (Applic <= Applicability.Not) {
+                Debug.Assert(false);
+                return new ErrorText("ERROR");
+            }
+            Debug.Assert(DataStream != null);
+
+            DataBuf = new byte[DataStream.Length];
+            DataStream.Position = 0;
+            DataStream.ReadExactly(DataBuf, 0, DataBuf.Length);
+
+            bool doMultirow = GetBoolOption(options, OPT_MULTIROW, false);
+            bool doSpacing = GetBoolOption(options, OPT_SPACING, false);
+
+            if (doMultirow) {
+                return ConvertIntoGrid(DataBuf, Palette8.Palette_MonoBW, doSpacing);
+            } else {
+                return Convert(DataBuf, Palette8.Palette_MonoBW, doSpacing);
+            }
+        }
+
+        /// <summary>
+        /// Converts a Fontrix font file to a B&amp;W bitmap.
+        /// </summary>
+        public Bitmap8 Convert(byte[] buf, Palette8 palette, bool doSpacing) {
+            Bitmap8 output;
+
+            int totalWidth = 0;
+            int maxHeight = 0;
+
+            int firstCharOffset = GetFirstCharacterOffset();
+            var charCount = GetCharacterCount();
+
+            for (int idx = firstCharOffset; idx < firstCharOffset + charCount; idx++) {
+
+                if (idx != firstCharOffset && doSpacing) {
+                    totalWidth += INTER_CHARACTER_SPACING_PX;
+                }
+
+                totalWidth += GetWidthForCharacter(idx);
+                int height = GetFontHeight();
+                if (height > maxHeight) { maxHeight = height; }
+
+            }
+
+            output = new Bitmap8(LEFT_PAD + totalWidth + RIGHT_PAD,
+                                 TOP_PADDING + maxHeight + BOTTOM_PADDING);
+
+            output.SetPalette(palette);
+
+            for (int idx = 0; idx < output.Width; idx++) {
```

Comment:
The other way to do this is to define your own palette and just make 0=white 1=black.  That lets you skip the fill and avoid the "1 - color" later on.

https://github.com/fadden/CiderPress2/blob/0edd3d6e90e0ff4335b3e77075f42169e8f897b6/FileConv/Gfx/HiRes_Font.cs#L48 defines a custom palette for that converter.  Note it uses 0=gray so that the border and spaces between entries are initialized to that and don't have to be drawn explicitly.  I'm not sure how that would look with proportional fonts though.

(You're also allowed to use a partially or fully transparent color, though PNG can't handle alpha with indexed color modes, so the PNG generator will switch to 32-bit direct color.)

We could also define a Palette_MonoWB in Palette8.cs and use that.

Reply:
I think that a WB or WBG palette is a good idea.  I will probably hold that back and integrate it as a separate PR that can affect both the Fontrix and Print Shop fonts after I get some of the current debugging issues that I'm slogging through at the moment.